File: sem2/po/po_proj2/src/studia/prog_ob/projekt2/world_simulator/organisms/animals/animals_subclasses/Fox.java
package studia.prog_ob.projekt2.world_simulator.organisms.animals.animals_subclasses;


import studia.prog_ob.projekt2.world_simulator.SimLogger;
import studia.prog_ob.projekt2.world_simulator.World;
import studia.prog_ob.projekt2.world_simulator.maps.MapLocation;
import studia.prog_ob.projekt2.world_simulator.organisms.Organism;
import studia.prog_ob.projekt2.world_simulator.organisms.animals.Animal;

public class Fox extends Animal {

    public static final int FOX_STRENGTH = 3;
    public static final int FOX_INITIATIVE = 7;
    public static final int FOX_SPEED = BASE_ANIMAL_SPEED;
    public static final char FOX_SYMBOL = 'F';

    public Fox(MapLocation new_map_location, World new_world) {
        super(FOX_STRENGTH, FOX_INITIATIVE, new_map_location, FOX_SYMBOL, new_world, FOX_SPEED);
    }

    @Override
    public Organism spawn_offspring(MapLocation location) {
        return getWorld().add_organism(new Fox(location, getWorld()));
    }

    @Override
    protected void pre_collision(Organism organismAtNewLocation) {
        if (organismAtNewLocation.getStrength() > this.getStrength()) {
            SimLogger.log(this + SimLogger.DANGERSENSE_MESSAGE);
            return;
        }

        super.pre_collision(organismAtNewLocation);
    }

    @Override
    public String toString() {
        return "Fox " + getMap_location() + " ";
    }
}

File: sem2/po/PO-VirtualWorld-Python/simulator/organisms/animals/Fox.py
from simulator.organisms.animals.Animal import Animal, BASE_ANIMAL_SPEED
from simulator.organisms.Organism import ORGANISM_TYPES

FOX_STRENGTH = 3
FOX_INITIATIVE = 7
FOX_SPEED = BASE_ANIMAL_SPEED


class Fox(Animal):
    def __init__(self, new_world, new_location):
        super().__init__(new_world, new_location, FOX_STRENGTH, FOX_INITIATIVE, ORGANISM_TYPES["Fox"], FOX_SPEED)

    def spawn_offspring(self, new_location):
        return Fox(self.world, new_location)

    def __str__(self):
        return "Fox at " + str(self.location)

    def pre_collision(self, autochton):
        if autochton.get_strength() > self.get_strength():
            self.tlog(" sensed " + str(autochton) + " and ran away")
        else:
            super().pre_collision(autochton)
